library/styleguideComponents/StyleGuide/ComponentsPage/index.js

```diff
@@ -0,0 +1,159 @@
+import GithubStars from '../../SectionHeading/GithubStarsWrapper';
+import Badge from '../../../../src/components/Badge';
+import githublogo from '../../SectionHeading/image/github.svg';
+import './styles.css';
+
+const GithubStarsBadge = GithubStars(({ stars }) => (
+    <Badge className="react-rainbow-github-badge rainbow-color_dark-1 rainbow-m-right_medium" variant="lightest">
+        <FontAwesomeIcon color="#061c3f" icon={faStar} pull="left" />
+        {stars}
+    </Badge>
+));
+
+export default class ComponentsPage extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            searchTerm: '',
+        };
+        this.handleOnChange = this.handleOnChange.bind(this);
+    }
+
+    getComponentsByCategory() {
+        const { searchTerm } = this.state;
+        const { components } = this.props;
+        const filtered = filterSectionsByName(components, searchTerm);
+        const categorizedComponents = { forms: [], layouts: [], dataViews: [], miscellaneous: [] };
+        filtered.forEach((component, index) => {
+            const key = `component-${index}`;
+            const src = `images/componentsThumbs/${component.name}.svg`;
+            const href = `/#/${component.name}`;
+            const card = (
+                <a href={href} key={key}>
+                    <Card
+                        className="react-rainbow-components-page_card"
+                        footer={component.name}>
+                        <img
+                            src={src}
+                            onError={this.handleOnError}
+                            className="react-rainbow-components-page_card-image"
+                            alt={`Component ${component.name}`} />
+                    </Card>
+                </a>
+            );
+            if (component.props.tags && component.props.tags.category) {
+                const category = component.props.tags.category[0].description;
+                if (category === 'Form') {
+                    categorizedComponents.forms.push(card);
+                } else if (category === 'Layout') {
+                    categorizedComponents.layouts.push(card);
+                } else if (category === 'DataView') {
+                    categorizedComponents.dataViews.push(card);
+                }
+            } else {
+                categorizedComponents.miscellaneous.push(card);
+            }
+        });
+        return categorizedComponents;
+    }
+
+    handleOnChange(event) {
+        this.setState({ searchTerm: event.target.value });
+    }
+
+    handleOnError(event) {
+        event.target.src = 'images/componentsThumbs/default-image.svg';
+    }
+
+    renderCategoryTitle(components, title) {
+        if (components.length) {
+            return (
+                <h1 className="react-rainbow-components-page_category-title">
+                    {title}
+                </h1>
+            );
+        }
+        return null;
```

Comment:
If this is not used here you can put this function outside the component.
I think is better to use RenderIf here. Instead of call:
```{this.renderCategoryTitle(layouts, 'Layout Components')}```
put directly a RenderIf with the condition.